tests/test_pyrflx.py

```diff
@@ -1033,6 +1021,26 @@ def test_array_typevalues(self) -> None:
         self.assertTrue(self.array_test_typeval.valid_message)
         self.assertEqual(self.array_test_typeval.bytestring, b"\x03\x05\x06\x07")
 
+    def test_array_preserve_value(self) -> None:
+        intval = IntegerValue(ModularInteger("Test.Int", Number(256)))
+        intval.assign(1)
+        enumval = EnumValue(
+            Enumeration(
+                "Test.Enum", {"something": Number(1), "other": Number(2)}, Number(2), False,
+            )
+        )
+        enumval.assign("something")
+        type_array = ArrayValue(Array("Test.Array", ModularInteger("Test.Mod_Int", Number(256))))
+        type_array.assign([intval])
+        self.assertEqual(type_array.value, [intval])
+        with self.assertRaisesRegex(
+            ValueError,
+            "cannot assign <class 'rflx.pyrflx.typevalue.EnumValue'>"
+            " to an array of <class 'rflx.model.ModularInteger'>",
```

Comment:
Can you change the error message to the class name instead of `<class ...>`?

Reply:
Fixed by 7f48161